report/validation/thermodesorption_spectra/dark/dark_tds.md

```diff
@@ -0,0 +1,256 @@
+
+To reproduce this experiment, six traps are needed: 1 intrinsic trap and 5 neutron induced traps.
+The trap densities for the neutron induced traps were fitted by {cite}`dark_modelling_2024` for each _dpa_ amount using FESTIM's `NeutronInducedTrap`.
+
+The damage distribution for the neutron-induced traps is as follows:
+$$
+    f(x) = \frac{1}{1 + \exp{ \frac{\left( x - x_0 \right)}{\Delta x} }}
+$$
+
+Tables with the relevant trap parameters are at the bottom of the page.
+
++++
+
+## FESTIM code
+
+```{code-cell} ipython3
+:tags: [hide-input, hide-output]
+
+# Setup
+import festim as F
+import numpy as np
+import sympy as sp
+import matplotlib.pyplot as plt
+
+# Parameters
+
+
+# Deuterium in Tungsten [35]
+D_0 = 1.6 * 1e-7 # m^2 s^-1
+E_D = 0.28 # eV
+w_atom_density = 6.3222e28
+
+sample_depth = 0.8e-3 # m
+
+# Table 2
+T_imp = 370 # K
+T_rest = 295 # K
+R_p = 0.7e-9 # m
+sigma = 0.5e-9 # m
+t_imp = 72 * 3600 # s
+t_rest = 12 * 3600 # s
+Beta = 0.05 # K s^-1
+
+fluence = 1.5e25
+flux = fluence / t_imp
+
+# Deuterium Beam Profile (S = flux * f(x))
+distribution = (1 / (sigma * (2 * np.pi) ** 0.5) * sp.exp(-0.5 * ((F.x - R_p) / sigma) ** 2))
+
+model = F.Simulation()
+
+vertices = np.concatenate(
+    [
+        np.linspace(0, 3e-9, num=100),
+        np.linspace(3e-9, 8e-6, num=100),
+        np.linspace(8e-6, 8e-5, num=100),
+        np.linspace(8e-5, 8e-4, num=100),
+    ]
+)
+model.mesh = F.MeshFromVertices(vertices)
+
+# material
+damaged_tungsten = F.Material(1, D_0, E_D, borders=[0, sample_depth])
+model.materials = damaged_tungsten
+
+# hydrogen source
+ion_flux = sp.Piecewise((flux*distribution, F.t < t_imp), (0, True))
+source_term = F.Source(value=ion_flux, volume=1, field=0)
+model.sources = [source_term]
+
+# boundary conditions
+model.boundary_conditions = [F.DirichletBC(surfaces=[1, 2], value=0, field="solute")]
+
+# temperature
+start_tds = t_imp + t_rest  # s
+min_temp, max_temp = 300, 1000
+
+model.T = F.Temperature(
+    value=sp.Piecewise(
+        (T_imp, F.t < t_imp),
+        (T_rest, F.t < start_tds),
+        (min_temp + Beta * (F.t - start_tds), True),
+    )
+)
+
+# trap settings
+k_0 = D_0 / (1.1e-10**2 * 6 * w_atom_density)
+damage_dist = 1 / (1 + sp.exp((F.x - 2.5e-06) / 5e-07))
+n_i = np.array([2.4e-3, 4.8, 3.8, 2.6, 3.6, 1.1]) * 1e25
+neutron_induced_traps = []
+E_p = [1.04, 1.15, 1.35, 1.65, 1.85, 2.05]
+
+trap_1 = F.Trap(
+    E_k = 0.28,
+    k_0 = k_0,
+    E_p = 1.04,
+    p_0 = 1e13,
+    density = n_i[0],
+    materials = damaged_tungsten
+)
+
+for i in range(1, 6):
+    neutron_induced_traps.append(F.Trap(
+        k_0 = k_0,
+        E_k = E_D,
+        p_0 = 1e13,
+        E_p = E_p[i],
+        density = n_i[i] * damage_dist,
+        materials = damaged_tungsten,
+    ))
+
+traps = [trap_1] + neutron_induced_traps
+model.traps = traps
+
+model.dt = F.Stepsize(
+    initial_value=1,
+    stepsize_change_ratio=1.1,
+    t_stop=t_imp + t_rest * 0.5,
+    dt_min=1e-1,
+    stepsize_stop_max=50,
```

Comment:
This triggers a deprecation warning:

```
/home/docs/checkouts/readthedocs.org/user_builds/festim-vv-report/conda/42/lib/python3.11/site-packages/festim/stepsize.py:54: DeprecationWarning: stepsize_stop_max and t_stop attributes will be deprecated in a future release, please use max_stepsize instead
  warnings.warn(
```

Instead please use the new `max_stepsize` parameter (see [here](https://festim.readthedocs.io/en/stable/userguide/stepsize.html#stepsize) for more details)